File: windowslib/src/main/resources/scala-native/windows/accCtrl/accessMode.c
#if defined(_WIN32) || defined(WIN32)
#define WIN32_LEAN_AND_MEAN
#include <accctrl.h>

int scalanative_not_used_access() { return NOT_USED_ACCESS; }
int scalanative_grant_access() { return GRANT_ACCESS; }
int scalanative_set_access() { return SET_ACCESS; }
int scalanative_deny_access() { return DENY_ACCESS; }
int scalanative_revoke_access() { return REVOKE_ACCESS; }
int scalanative_set_audit_success() { return SET_AUDIT_SUCCESS; }
int scalanative_set_audit_failure() { return SET_AUDIT_FAILURE; }

#endif // defined(_WIN32)

File: windowslib/src/main/scala/scala/scalanative/windows/accctrl/AccessMode.scala
package scala.scalanative.windows.accctrl

import scala.scalanative.unsafe.*

@extern
object AccessMode {
  @name("scalanative_not_used_access")
  def NOT_USED_ACCESS: AccessMode = extern
  @name("scalanative_grant_access")
  def GRANT_ACCESS: AccessMode = extern
  @name("scalanative_set_access")
  def SET_ACCESS: AccessMode = extern
  @name("scalanative_deny_access")
  def DENY_ACCESS: AccessMode = extern
  @name("scalanative_revoke_access")
  def REVOKE_ACCESS: AccessMode = extern
  @name("scalanative_set_audit_success")
  def SET_AUDIT_SUCCESS: AccessMode = extern
  @name("scalanative_set_audit_failure")
  def SET_AUDIT_FAILURE: AccessMode = extern
}
